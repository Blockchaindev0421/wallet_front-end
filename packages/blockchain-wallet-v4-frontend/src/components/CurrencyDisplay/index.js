import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { convertBaseCoinToFiat, displayWeiToFiat } from 'services/ConversionService'
import { selectors } from 'data'

const CurrencyDisplay = props => {
  const { currency, bitcoinRates, ethereumRates, coin, children } = props
  const amount = children || '0'
  return (
    <div>
      { coin === 'BTC'
        ? convertBaseCoinToFiat(currency, bitcoinRates, amount)
        : displayWeiToFiat(currency, ethereumRates, amount)
      }
    </div>
  )
}

CurrencyDisplay.propTypes = {
  coin: PropTypes.oneOf(['BTC', 'ETH']).isRequired,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
}

CurrencyDisplay.defaultProps = {
  children: 0
}

const mapStateToProps = (state) => ({
  currency: selectors.core.settings.getCurrency(state),
  bitcoinRates: selectors.core.data.rates.getBtcRates(state),
  ethereumRates: selectors.core.data.rates.getEthRates(state)
})

export default connect(mapStateToProps)(CurrencyDisplay)
