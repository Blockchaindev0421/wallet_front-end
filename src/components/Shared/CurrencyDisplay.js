import React from 'react'
import PropTypes from 'prop-types'
import {connect} from 'react-redux'

import { convertToCurrency } from 'services/conversionService'
import { selectors } from 'data'

export const CurrencyDisplay = ({ className, value }) => (
  <span className={className}>{value}</span>
)

CurrencyDisplay.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string,
  element: PropTypes.string
}

const CurrencyDisplayContainer = (props) => {
  let conversion = convertToCurrency(props.amount, props.currency, props.rates).getOrElse('N/A')
  return (<CurrencyDisplay className={props.className} value={conversion} />)
}

CurrencyDisplayContainer.propTypes = {
  amount: PropTypes.number.isRequired
}

let mapStateToProps = (state) => ({
  currency: selectors.core.settings.getCurrency(state),
  rates: selectors.core.rates.getRates(state)
})

export default connect(mapStateToProps)(CurrencyDisplayContainer)
