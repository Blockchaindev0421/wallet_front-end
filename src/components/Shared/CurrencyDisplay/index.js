import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import { convertCoinToFiat } from 'services/ConversionService'
import { selectors } from 'data'
import { Typography } from 'components/generic/Typography'

const CurrencyDisplay = ({ ...props, children }) => {
  const { network, currency, rates, ...rest } = props
  const fiat = convertCoinToFiat(network, children, currency, rates).getOrElse({ amount: 'N/A' })

  return (
    <Typography {...rest}>{`${fiat.symbol} ${fiat.amount.toFixed(2)}`}</Typography>)
}

CurrencyDisplay.propTypes = {
  children: PropTypes.number.isRequired
}

CurrencyDisplay.defaultProps = {
  children: 0
}

const mapStateToProps = (state) => ({
  network: 'bitcoin',
  currency: selectors.core.settings.getCurrency(state),
  rates: selectors.core.rates.getRates(state)
})

export default connect(mapStateToProps)(CurrencyDisplay)
