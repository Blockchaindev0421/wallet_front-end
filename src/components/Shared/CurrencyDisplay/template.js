import React from 'react'
import PropTypes from 'prop-types'

const CurrencyDisplay = (props) => {
  return (
    <span className={props.className}>{props.value}</span>
  )
}

CurrencyDisplay.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string
}

export default CurrencyDisplay
