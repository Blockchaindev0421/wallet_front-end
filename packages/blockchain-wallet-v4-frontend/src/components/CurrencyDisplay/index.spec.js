import React from 'react'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

import CurrencyDisplay from './index.js'

describe('CurrencyDisplay', () => {
  const mockStore = configureStore()

  test('Displays 100000000 satoshis as local currency', () => {
    let initialState = {
      settings: {
        currency: 'GBP'
      },
      data: {
        rates: { 'GBP': { last: 2000, symbol: 'Â£' } }
      }
    }

    let store = mockStore(initialState)
    let component = renderer.create(
      <CurrencyDisplay store={store} className='button'>{100000000}</CurrencyDisplay>
    )

    expect(component.toJSON()).toMatchSnapshot()
  })
})
