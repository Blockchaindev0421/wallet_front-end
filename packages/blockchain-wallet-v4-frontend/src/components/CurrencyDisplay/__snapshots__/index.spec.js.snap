// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyDisplay Displays 100000000 satoshis as local currency 1`] = `
<div
  className="button sc-bdVaJa kwhmts"
>
  N/A
</div>
`;
